### PR TITLE
[sglang] feat: Add multi-interaction registry support and testing

### PR DESCRIPTION
**Pull request by SwordFaith**
### What does this PR do?

> Add **concise** overview of what this PR aims to achieve or accomplish. Reference related GitHub issues and PRs that help with the review.

  This PR implements multi-interaction support in SGLangRollout, enabling sample-level interaction selection similar to the existing tools system. The implementation includes a new interaction registry system that allows multiple named
  interactions to be configured and used within a single rollout instance. #1630 

Core Implementation

  - New Interaction Registry System: Created verl/interactions/utils/interaction_registry.py with functions to dynamically load and manage multiple interaction instances from configuration files
  - Enhanced SGLangRollout:
    - Replaced single interaction attribute with interaction_map: dict[str, BaseInteraction]
    - Updated _initialize_interactions() method to support multiple interactions via registry
    - Modified interaction selection logic to use interaction_kwargs.name for sample-level binding
  - Configuration Updates: Added name field support in interaction config format with automatic name generation fallback

  Data Processing

  - Updated GSM8K Preprocessing: Modified examples/data_preprocess/gsm8k_multiturn_w_interaction.py to inject name field in interaction_kwargs
  - Enhanced Configuration: Updated examples/sglang_multiturn/config/interaction_config/gsm8k_interaction_config.yaml with explicit name field

  Testing & Quality

  - Comprehensive Test Suite: Added tests/interactions/test_interaction_registry.py with full coverage of registry functionality
  - Integration Tests: Created tests/workers/rollout/test_sglang_multi_interaction.py for multi-interaction scenarios
  - Updated Existing Tests: Modified existing interaction tests to support new name attribute and configuration format
  - Error Handling: Added validation for duplicate names, missing interactions, and edge cases

  Backward Compatibility

  - Graceful Degradation: When no interaction config is provided, system works without interactions (empty interaction_map)
  - Default Name Handling: Falls back to "gsm8k" when no name is specified in interaction_kwargs
  - Existing API Preservation: All existing interaction functionality remains unchanged

 Key Features

  1. Sample-Level Selection: Each sample can specify which interaction to use via interaction_kwargs.name
  2. Registry Pattern: Similar architecture to existing tools system for consistency
  3. Automatic Naming: Intelligent name generation from class names (e.g., Gsm8kInteraction → gsm8k)
  4. Duplicate Prevention: Runtime validation prevents naming conflicts
  5. Flexible Configuration: Supports both explicit names and automatic derivation

### Checklist Before Starting

- [ ] Search for similar PRs. Paste at least one query link here: ...
- [ ] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)
  - `{modules}` include `fsdp`, `megatron`, `sglang`, `vllm`, `rollout`, `trainer`, `ci`, `training_utils`, `recipe`, `hardware`, `deployment`, `ray`, `worker`, `single_controller`, `misc`, `perf`, `model`, `algo`, `env`, `tool`, `ckpt`, `doc`, `data`
  - If this PR involves multiple modules, separate them with `,` like `[megatron, fsdp, doc]`
  - `{type}` is in `feat`, `fix`, `refactor`, `chore`, `test`
  - If this PR breaks any API (CLI arguments, config, function signature, etc.), add `[BREAKING]` to the beginning of the title.
  - Example: `[BREAKING][fsdp, megatron] feat: dynamic batching`

### Test

> For changes that can not be tested by CI (e.g., algorithm implementation, new model support), validate by experiment(s) and show results like training curve plots, evaluation results, etc.

  - Unit tests for interaction registry functionality
  - Integration tests for multi-interaction rollout scenarios
  - Backward compatibility tests for existing single-interaction workflows
  - Error handling tests for edge cases and invalid configurations
  - Updated existing GSM8K interaction tests

### API and Usage Example

> Demonstrate how the API changes if any, and provide usage example(s) if possible.

 Configuration (interaction_config.yaml):
  interaction:
    - name: "gsm8k"
      class_name: "verl.interactions.gsm8k_interaction.Gsm8kInteraction"
      config: {}
    - name: "custom_solver"
      class_name: "custom.interactions.CustomInteraction"
      config: {"solver_type": "advanced"}

  Data Sample:

```python
  {
      "interaction_kwargs": {
          "name": "gsm8k",
          "query": "What is 2+2?",
          "ground_truth": "4"
      }
  }
```

### High-Level Design

> Demonstrate the high-level design if this PR is complex.

### Specific Changes

> List the specific changes.

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [x] Read the [Contribute Guide](https://github.com/volcengine/verl?tab=readme-ov-file#contribution-guide).
- [x] Apply [pre-commit checks](https://github.com/volcengine/verl?tab=readme-ov-file#code-linting-and-formatting): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [ ] Add / Update [the documentation](https://github.com/volcengine/verl/tree/main/docs).
- [x] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/volcengine/verl/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...
- [x] Once your PR is ready for CI, send a message in [the `ci-request` channel](https://verl-project.slack.com/archives/C091TCESWB1) in [the `verl` Slack workspace](https://join.slack.com/t/verl-project/shared_invite/zt-3855yhg8g-CTkqXu~hKojPCmo7k_yXTQ).

### CODE REVIEW

File: verl/interactions/utils/interaction_registry.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import os
+import sys
+
+from omegaconf import OmegaConf
+
+logger = logging.getLogger(__file__)
+logger.setLevel(os.getenv("VERL_LOGGING_LEVEL", "WARN"))
+
+
+def get_interaction_class(cls_name):
```
Comment (chenhaiq) [inline verl/interactions/utils/interaction_registry.py:27]: 可以复用已有的 verl/tools/utils/tool_registry.py::get_tool_class 函数

File: verl/interactions/utils/interaction_registry.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        sys.modules[module_name] = module
+        spec.loader.exec_module(module)
+    else:
+        module = sys.modules[module_name]
+
+    interaction_cls = getattr(module, class_name)
+    return interaction_cls
+
+
+def initialize_interactions_from_config(interaction_config_file):
```
Comment (chenhaiq) [inline verl/interactions/utils/interaction_registry.py:42]: 跟已有的 verl/tools/utils/tool_registry.py::initialize_tools_from_config 也有重复，能不能重构下已有的那个函数，让mcp也使用map的新格式？

File: verl/workers/rollout/sglang_rollout/sglang_rollout.py [hunk 3]
```diff
@@ -926,9 +924,15 @@ async def _handle_pending_state(self, _req: AsyncRolloutRequest) -> AsyncRollout
                 create_kwargs = _req.tools_kwargs[tool.name].get("create_kwargs", {})
                 tool_creation_coroutines.append(tool.create(_req.request_id, **create_kwargs))
             await asyncio.gather(*tool_creation_coroutines)
-        if _req.interaction_kwargs:
+        if _req.interaction_kwargs and self.interaction_map:
             interaction_kwargs = _req.interaction_kwargs
-            await self.interaction.start_interaction(_req.request_id, **interaction_kwargs)
```
Comment (chenhaiq) [inline verl/workers/rollout/sglang_rollout/sglang_rollout.py:931]: 跟上面的代码也是重复的，可以合并成  get_ interaction 方法

File: verl/interactions/utils/interaction_registry.py (hunk 1 above)
Comment (SwordFaith) [reply verl/interactions/utils/interaction_registry.py:27]: interaction and tool are different concepts. Interactions are playing user role which used in partial exposure/collaborative task, reflection task and workflow node training task.
